src/zope/configuration/config.py

```diff
@@ -769,29 +764,31 @@ def execute_actions(self, clear=True, testing=False):
                 info = action['info']
                 try:
                     callable(*args, **kw)
-                except pass_through_exceptions:
-                    raise
-                except Exception:
-                    t, v, tb = sys.exc_info()
+                except BaseException as ex:
                     try:
-                        reraise(ConfigurationExecutionError(t, v, info),
-                                None, tb)
+                        if isinstance(ex, ConfigurationError):
+                            ex.append_details(info)
+                            raise
+                        if isinstance(ex, pass_through_exceptions):
+                            raise
+                        if not isinstance(ex, Exception):
+                            # BaseException
+                            raise
```

Comment:
I was doing this in one catch block to avoid duplicating the `if isinstance(ex, ConfigurationError): ex.append_details(info); raise` logic. ConfigurationError could be caught by both `pass_through_exceptions` and `Exception`.

But I see now that I could just list `except ConfigurationError` *first* and solve the problem.